### DISCUSSION TITLE
No precision and recall metrics?

### DISCUSSION THREAD

**Discussion by Allen2019** [chemprop/chemprop#850] [Q&A]
When I try to specify "--metric precision" for training/cross-validating a classification model, I get the error:

> ArgumentTypeError: Value for variable "metric" must be one of ['auc', 'prc-auc', 'rmse', 'mae', 'mse', 'r2', 'accuracy', 'cross_entropy', 'binary_cross_entropy', 'sid', 'wasserstein', 'f1', 'mcc', 'bounded_rmse', 'bounded_mae', 'bounded_mse'].

Even though, in the [documentation](https://github.com/chemprop/chemprop/blob/master/README.md), it says that "auc (default), prc-auc, accuracy, binary_cross_entropy, f1, mcc, recall, precision and balanced accuracy" are available for classification.

So, why the discrepancy?

Here is my code:

> import chemprop
> 
> arguments = [
>     '--data_path', 'data_no_index.csv',
>     '--dataset_type', 'classification',
>     '--save_dir', 'Checkpoints_2',
>     '--num_folds','5',
>     '--config_path', 'hyperparameters.json',
>     '--metric', 'precision',
>     '--extra_metrics', 'recall'
> ]
> 
> args = chemprop.args.TrainArgs().parse_args(arguments)

Thanks!

**Comment by kevingreenman** [comment]
Which version of chemprop are you using? Precision was added in #542, which was not included in a release until v1.7.0. Can you try upgrading chemprop to 1.7.0 or 1.7.1?
